web/src/lib/components/shared-components/profile-image-cropper.svelte

```diff
@@ -0,0 +1,91 @@
+<script lang="ts">
+  import { AssetResponseDto, api } from '@api';
+  import { createEventDispatcher } from 'svelte';
+  import { notificationController, NotificationType } from './notification/notification';
+  import { handleError } from '$lib/utils/handle-error';
+  import domtoimage from 'dom-to-image';
+  import PhotoViewer from '../asset-viewer/photo-viewer.svelte';
+  import BaseModal from './base-modal.svelte';
+  import Button from '../elements/buttons/button.svelte';
+
+  export let asset: AssetResponseDto;
+  export let publicSharedKey = '';
+
+  const dispatch = createEventDispatcher();
+  let imgElement: HTMLDivElement;
+
+  const hasTransparentPixels = async (blob: Blob) => {
+    const img = new Image();
+    img.src = URL.createObjectURL(blob);
+    await img.decode();
+    return new Promise((resolve) => {
+      img.onload = function () {
```

Comment:
This promise and callback are not needed with `img.decode()`.

Reply:
thanks, sry, fixed now